src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAccountManager.java

```diff
@@ -1999,4 +1999,49 @@ public Map<SchoolInfoStatus, Long> getSchoolInfoStats() throws SegueDatabaseExce
   public Long getNumberOfAnonymousUsers() throws SegueDatabaseException {
     return temporaryUserCache.getCountOfAnonymousUsers();
   }
+
+  public RegisteredUserDTO updateTeacherPendingFlag(final Long userId, boolean newFlagValue)
+      throws SegueDatabaseException, NoUserException {
+    RegisteredUser user = findUserById(userId);
+    if (user == null) {
+      throw new NoUserException("No user found with this ID.");
+    }
+    user.setTeacherPending(newFlagValue);
+    RegisteredUser updatedUser = database.createOrUpdateUser(user);
+    return dtoMapper.map(updatedUser, RegisteredUserDTO.class);
+  }
+
+  public void sendRoleChangeRequestEmail(final HttpServletRequest request, final RegisteredUserDTO currentUser,
```

Comment:
Small one, but in the context of this method, `currentUser` should be simply called `user`, because the method does not know anything about the status of the user, it only seems to care about it being a user.

Reply:
Oh, fair point. The only _current_ usage is with the currently logged-in user, retrieved from the database, but there's no reason the function couldn't be called in another situation...

Reply:
Yes, what I tend to do is that I normally analyse functions in isolation. In this case, by naming this parameter `currentUser` you're giving away the current way the method is used outside the context of this class. So you're almost very subtly breaking its encapsulation, I think. But in the context of a method, the method shouldn't be concerned with how it's being called, if that makes sense.